src/librbd/exclusive_lock/PreReleaseRequest.cc

```diff
@@ -301,6 +295,9 @@ void PreReleaseRequest<I>::send_unlock() {
 
 template <typename I>
 void PreReleaseRequest<I>::finish() {
+  if (m_error_result < 0) {
```

Comment:
We cannot release the lock but then allow write operations to proceed

Reply:
I just move unblock_writes here. The logic is not changed. It is released in the old logic already.

By the way, as the lock is shutting down, which is called when closing image, there should not any write operations here and after.

Reply:
... that's only true on a shut down case. You can release the lock w/o shutting down.